src/git/GitRepository.h

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2023 Nexus6 <nexus6.haiku@icloud.com>
+ * All rights reserved. Distributed under the terms of the MIT license.
+ * Based on TrackGit (https://github.com/HaikuArchives/TrackGit)
+ * Original author: Hrishikesh Hiraskar <hrishihiraskar@gmail.com>
+ * Copyright Hrishikesh Hiraskar and other HaikuArchives contributors (see GitHub repo for details)
+ */
+
+#pragma once
+
+#include <Catalog.h>
+#include <SupportDefs.h>
+#include <Notification.h>
+
+#include <git2.h>
+
+#include <functional>
+#include <string>
+#include <vector>
+
+#include "GException.h"
+#include "Log.h"
+
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "GitRepository"
+
+namespace Genio::Git {
+
+	const int CANCEL_CREDENTIALS = -123;
+
+	enum PullResult {
+		UpToDate,
+		FastForwarded,
+		Merged
+	};
+
+	// Git exceptions
+
+	class GitException : public GException {
+	public:
+
+		GitException(int error, BString const& message)
+			:
+			GException(error, message),
+			_error(error),
+			_message(message)
+		{
+			LogError("GitException %s, error = %d" , message.String(), error);
+		}
+
+		int							Error() const noexcept { return _error; }
```

Comment:
Doesn't GException have these method already? I think we're duplicating them here

Reply:
yes, you are right. Done